### ISSUE TITLE
Stack overflow in AddQueryIdAndResultObject (ClientRequest.cs, line 548)

### ISSUE THREAD

**Issue by Dr-Oli** [SharePoint/sp-dev-docs#8633]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

SharePoint CSOM

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [X] other (enter in the "Additional environment details" area below)

### Additional environment details

Desktop application based on .NET 4.7.2 on Windows 10

### Describe the bug / error

From time to time when running file.ListItemAllFields my application is crashing. In dump analyses I am finding:

************* Path validation summary **************
Response                         Time (ms)     Location
Deferred                                       srv*
Symbol search path is: srv*
Executable search path is: 
Windows 10 Version 19044 MP (8 procs) Free x64
Product: WinNt, suite: SingleUserTS
Edition build lab: 19041.1.amd64fre.vb_release.191206-1406
Machine Name:
Debug session time: Mon Dec  5 10:58:45.000 2022 (UTC + 1:00)
System Uptime: 3 days 12:28:43.246
Process Uptime: 0 days 0:31:44.000
................................................................
................................................................
................................................................
.......................................................
Loading unloaded module list
........
This dump file has an exception of interest stored in it.
The stored exception information can be accessed via .ecxr.
(6904.4ea4): Stack overflow - code c00000fd (first/second chance not available)

+------------------------------------------------------------------------+
| This target supports Hardware-enforced Stack Protection. A HW based    |
| "Shadow Stack" may be available to assist in debugging and analysis.   |
| See aka.ms/userhsp for more info.                                      |
|                                                                        |
| dps @ssp                                                               |
|                                                                        |
+------------------------------------------------------------------------+

For analysis of this file, run !analyze -v
clr!SlowAllocateString+0x3a:
00007ff8`a45ff2fe e89de50600      call    clr!GetThread (00007ff8`a466d8a0)
0:023> !analyze -v
*******************************************************************************
*                                                                             *
*                        Exception Analysis                                   *
*                                                                             *
*******************************************************************************

MethodDesc:   00007ff8470f71c0
Method Name:  Microsoft.SharePoint.Client.ClientRequest.AddQueryIdAndResultObject(Int64, System.Object)
Class:        00007ff8470ef838
MethodTable:  00007ff8470f7358
mdToken:      00000000060000db
Module:       00007ff845314428
IsJitted:     yes
CodeAddr:     00007ff846c0e060
Transparency: Transparent
MethodDesc:   00007ff845315f10
Method Name:  Microsoft.SharePoint.Client.ClientObject.get_Query()
Class:        00007ff84534cdf0
MethodTable:  00007ff845315fd8
mdToken:      000000000600002c
Module:       00007ff845314428
IsJitted:     yes
CodeAddr:     00007ff846c0eee0
Transparency: Transparent
MethodDesc:   00007ff847124b18
Method Name:  Microsoft.SharePoint.Client.ListItem.LoadExpandoFields()
Class:        00007ff84710b268
MethodTable:  00007ff847124cc8
mdToken:      0000000006000602
Module:       00007ff845311e08
IsJitted:     yes
CodeAddr:     00007ff848f26780
Transparency: Transparent
MethodDesc:   00007ff845315f10
Method Name:  Microsoft.SharePoint.Client.ClientObject.get_Query()
Class:        00007ff84534cdf0
MethodTable:  00007ff845315fd8
mdToken:      000000000600002c
Module:       00007ff845314428
IsJitted:     yes
CodeAddr:     00007ff846c0eee0
Transparency: Transparent
MethodDesc:   00007ff847124b18
Method Name:  Microsoft.SharePoint.Client.ListItem.LoadExpandoFields()
Class:        00007ff84710b268
MethodTable:  00007ff847124cc8
mdToken:      0000000006000602
Module:       00007ff845311e08
IsJitted:     yes
CodeAddr:     00007ff848f26780
Transparency: Transparent
MethodDesc:   00007ff845315f10
Method Name:  Microsoft.SharePoint.Client.ClientObject.get_Query()
Class:        00007ff84534cdf0
MethodTable:  00007ff845315fd8
mdToken:      000000000600002c
Module:       00007ff845314428
IsJitted:     yes
CodeAddr:     00007ff846c0eee0
Transparency: Transparent
MethodDesc:   00007ff847124b18
Method Name:  Microsoft.SharePoint.Client.ListItem.LoadExpandoFields()
Class:        00007ff84710b268
MethodTable:  00007ff847124cc8
mdToken:      0000000006000602
Module:       00007ff845311e08
IsJitted:     yes
CodeAddr:     00007ff848f26780
Transparency: Transparent
MethodDesc:   00007ff845315f10
Method Name:  Microsoft.SharePoint.Client.ClientObject.get_Query()
Class:        00007ff84534cdf0
MethodTable:  00007ff845315fd8
mdToken:      000000000600002c
Module:       00007ff845314428
IsJitted:     yes
CodeAddr:     00007ff846c0eee0
Transparency: Transparent
MethodDesc:   00007ff847124b18
Method Name:  Microsoft.SharePoint.Client.ListItem.LoadExpandoFields()
Class:        00007ff84710b268
MethodTable:  00007ff847124cc8
mdToken:      0000000006000602
Module:       00007ff845311e08
IsJitted:     yes
CodeAddr:     00007ff848f26780
Transparency: Transparent
MethodDesc:   00007ff845315f10
Method Name:  Microsoft.SharePoint.Client.ClientObject.get_Query()
Class:        00007ff84534cdf0
MethodTable:  00007ff845315fd8
mdToken:      000000000600002c
Module:       00007ff845314428
IsJitted:     yes
CodeAddr:     00007ff846c0eee0
Transparency: Transparent
MethodDesc:   00007ff847124b18
Method Name:  Microsoft.SharePoint.Client.ListItem.LoadExpandoFields()
Class:        00007ff84710b268
MethodTable:  00007ff847124cc8
mdToken:      0000000006000602
Module:       00007ff845311e08
IsJitted:     yes
CodeAddr:     00007ff848f26780
Transparency: Transparent
MethodDesc:   00007ff845315f10
Method Name:  Microsoft.SharePoint.Client.ClientObject.get_Query()
Class:        00007ff84534cdf0
MethodTable:  00007ff845315fd8
mdToken:      000000000600002c
Module:       00007ff845314428
IsJitted:     yes
CodeAddr:     00007ff846c0eee0
Transparency: Transparent

KEY_VALUES_STRING: 1

    Key  : Analysis.CPU.mSec
    Value: 125031

    Key  : Analysis.DebugAnalysisManager
    Value: Create

    Key  : Analysis.Elapsed.mSec
    Value: 144326

    Key  : Analysis.IO.Other.Mb
    Value: 1

    Key  : Analysis.IO.Read.Mb
    Value: 1

    Key  : Analysis.IO.Write.Mb
    Value: 91

    Key  : Analysis.Init.CPU.mSec
    Value: 1812

    Key  : Analysis.Init.Elapsed.mSec
    Value: 32658

    Key  : Analysis.Memory.CommitPeak.Mb
    Value: 901

    Key  : CLR.BuiltBy
    Value: NET481REL1LAST_C

    Key  : CLR.Engine
    Value: CLR

    Key  : CLR.Version
    Value: 4.8.9105.0

    Key  : Timeline.OS.Boot.DeltaSec
    Value: 304123

    Key  : Timeline.Process.Start.DeltaSec
    Value: 1904

    Key  : WER.OS.Branch
    Value: vb_release

    Key  : WER.OS.Timestamp
    Value: 2019-12-06T14:06:00Z

    Key  : WER.OS.Version
    Value: 10.0.19041.1

    Key  : WER.Process.Version
    Value: 2022.4.10.0

....

ERROR_CODE: (NTSTATUS) 0xc00000fd - A new guard page for the stack cannot be created.

### Steps to reproduce

This is tricky part as this is happening randomly. Usually (90%) it works but then same files are processed again and it is crashing.
Dump file can be shared (it is around 1,5Gb).

### Expected behavior

ListItemAllFields should list all fields of checked file without going into endless loop ending with stack overflow error.

**Comment by Dr-Oli** [comment]
Hi Alex, 

any news here? Any chance to get some estimate when somebody will be assigned?

Thx

**Comment by Dr-Oli** [comment]
I managed to remove ListItemAllFields but still there is a problem. From Windbg I see after running !analyze -v

....
00000098`81009730 00007ff8`d163ca20 Microsoft_SharePoint_Client_Runtime!Microsoft.SharePoint.Client.ClientObject.Retrieve+0x10
00000098`81009780 00007ff8`d163c9ca Microsoft_SharePoint_Client!Microsoft.SharePoint.Client.ListItem.LoadExpandoFields+0x11a
00000098`810097e0 00007ff8`d112a311 Microsoft_SharePoint_Client_Runtime!Microsoft.SharePoint.Client.ClientObject.get_Query+0x2b1
00000098`81009830 00007ff8`d163ca20 Microsoft_SharePoint_Client_Runtime!Microsoft.SharePoint.Client.ClientObject.Retrieve+0x10
00000098`81009880 00007ff8`d163c9ca Microsoft_SharePoint_Client!Microsoft.SharePoint.Client.ListItem.LoadExpandoFields+0x11a
00000098`810098e0 00007ff8`d112a311 Microsoft_SharePoint_Client_Runtime!Microsoft.SharePoint.Client.ClientObject.get_Query+0x2b1
00000098`81009930 00007ff8`d163ca20 Microsoft_SharePoint_Client_Runtime!Microsoft.SharePoint.Client.ClientObject.Retrieve+0x10
00000098`81009980 00007ff8`d163c9ca Microsoft_SharePoint_Client!Microsoft.SharePoint.Client.ListItem.LoadExpandoFields+0x11a
00000098`810099e0 00007ff8`d112a311 Microsoft_SharePoint_Client_Runtime!Microsoft.SharePoint.Client.ClientObject.get_Query+0x2b1
00000098`81009a30 00007ff8`d163ca20 Microsoft_SharePoint_Client_Runtime!Microsoft.SharePoint.Client.ClientObject.Retrieve+0x10
00000098`81009a80 00007ff8`d163c9ca Microsoft_SharePoint_Client!Microsoft.SharePoint.Client.ListItem.LoadExpandoFields+0x11a
00000098`81009ae0 00007ff8`d112a311 Microsoft_SharePoint_Client_Runtime!Microsoft.S

STACK_COMMAND:  ** Pseudo Context ** ManagedPseudo ** Value: ffffffff ** ; kb

SYMBOL_NAME:  Microsoft_SharePoint_Client_Runtime!Microsoft.SharePoint.Client.ClientQueryInternal.Select+58

MODULE_NAME: Microsoft_SharePoint_Client_Runtime

IMAGE_NAME:  Microsoft.SharePoint.Client.Runtime.dll

FAILURE_BUCKET_ID:  CLR_EXCEPTION_STACK_OVERFLOW_c00000fd_Microsoft.SharePoint.Client.Runtime.dll!Microsoft.SharePoint.Client.ClientQueryInternal.Select

OS_VERSION:  10.0.19041.1

BUILDLAB_STR:  vb_release

OSPLATFORM_TYPE:  x64

OSNAME:  Windows 10

IMAGE_VERSION:  16.1.23508.12000

FAILURE_ID_HASH:  {f501fe81-72e2-b4f6-2b24-1990e81a77f0}

Followup:     MachineOwner

**Comment by Dr-Oli** [comment]
If I open dump in Visual Studio I see that 3 frames are repeated 16k times and certainly I do not have that in my code:

![image](https://user-images.githubusercontent.com/55049779/229478341-113eb11c-3ccb-465d-98e2-5d7ae5a25412.png)

**Comment by Dr-Oli** [comment]
If I go deeper in dump file I see that Microsoft.SharePoint.Client.ClientQueryProperty has 263066 elements which does not make sense and they l 32 bytes long:

||2:2:041> !DumpObj /d 000001b0112d7188
Name:        Microsoft.SharePoint.Client.ClientQueryProperty
MethodTable: 00007ff8d145baf0
EEClass:     00007ff8d1461628
Size:        32(0x20) bytes
File:        C:\Program Files (x86)\ReplaceMagic\Microsoft.SharePoint.Client.Runtime.dll
Fields:
              MT    Field   Offset                 Type VT     Attr            Value Name
00007ff92d10c590  4000033       10       System.Boolean  1 instance                1 ScalarProperty
00007ff92d10c590  4000034       11       System.Boolean  1 instance                1 ScalarPropertySet
00007ff92d10c590  4000035       12       System.Boolean  1 instance                0 SelectAll
00007ff92d10c590  4000036       13       System.Boolean  1 instance                0 SelectAllSet
00007ff8d1459fc0  4000037        8 ...ientQueryInternal  0 instance 0000000000000000 Query

and this is how they are pulled. 
To me it looks like that somehow Microsoft.SharePoint.Client is having recursion which is ended in endless loop.

**Comment by Dr-Oli** [comment]
Btw. application is processing a lot of document concurrently. Each of document is getting own ClientContext which is created like:
docContext = mainContext.Clone(URI)
to avoid problems with multiple threads. At least, this is what I found.
If this is correct and docContext is really now thread-safe, there has to be some problem with ClientRequest.XXXX when there are a lot in same location (can similar name be a problem)?

**Comment by Dr-Oli** [comment]
Had to change title as I think that this is not connected to ListItemAllFields but for some reasons when there is concurent and long processing of documents (we need to download documents, make changes and upload them back) something is happening and this is leading to too many repeated executions:

![image](https://user-images.githubusercontent.com/55049779/230136361-89f306fe-4787-4b4c-9ff2-7dd947461642.png)

which does not make any sense as there are no more than 3-4 documents processed in parallel and all ClientContext are created for each file as Clone of initial ClientContext. Any chance to get somehow Microsoft.SharePoint.Client.Runtime.pdb to check during debug of crash dump?

Not sure if this is helpful but from view disassembly:

00007FF8D0FD315A  mov         r8,7FF8CF73E060h  
00007FF8D0FD3164  mov         r9d,1  
00007FF8D0FD316A  call        00007FF92ED85940  
00007FF8D0FD316F  cmp         qword ptr [rsi+38h],0  
00007FF8D0FD3174  jne         Microsoft.SharePoint.Client.ClientRuntimeContext.AddQueryIdAndResultObject(Int64, System.Object)+06Fh (07FF8D0FD319Fh)  
00007FF8D0FD3176  mov         rcx,7FF8D124DCD0h  
00007FF8D0FD3180  call        CORINFO_HELP_NEWSFAST (07FF92EA8FFB0h)  
00007FF8D0FD3185  mov         rbp,rax  
00007FF8D0FD3188  mov         rcx,rbp  
00007FF8D0FD318B  mov         rdx,rsi  
00007FF8D0FD318E  call        Microsoft.SharePoint.Client.ClientRequest..ctor(Microsoft.SharePoint.Client.ClientRuntimeContext) (07FF8D0FD2A10h)  
00007FF8D0FD3193  lea         rcx,[rsi+38h]  
00007FF8D0FD3197  mov         rdx,rbp  
00007FF8D0FD319A  call        CORINFO_HELP_ASSIGN_REF (07FF92EA8DC60h)  
00007FF8D0FD319F  mov         rcx,qword ptr [rsi+38h]  
00007FF8D0FD31A3  mov         rdx,rbx  
00007FF8D0FD31A6  mov         r8,rdi  
00007FF8D0FD31A9  mov         rax,offset Microsoft.SharePoint.Client.ClientRequest.AddQueryIdAndResultObject(Int64, System.Object) (07FF8D0FD1300h)  
00007FF8D0FD31B3  cmp         dword ptr [rcx],ecx  
00007FF8D0FD31B5  add         rsp,28h  
00007FF8D0FD31B9  pop         rbx  
00007FF8D0FD31BA  pop         rbp  
00007FF8D0FD31BB  pop         rsi  
00007FF8D0FD31BC  pop         rdi  
00007FF8D0FD31BD  jmp         rax  
00007FF8D0FD31C0  mov         rcx,7FF92D123F90h  
00007FF8D0FD31CA  call        CORINFO_HELP_NEWSFAST (07FF92EA8FFB0h)  
00007FF8D0FD31CF  mov         rsi,rax  
00007FF8D0FD31D2  mov         ecx,4Dh  
00007FF8D0FD31D7  mov         rdx,7FF8CF73CBB8h  
00007FF8D0FD31E1  call        00007FF92EA87F00  
00007FF8D0FD31E6  mov         rdx,rax  
00007FF8D0FD31E9  mov         rcx,rsi  
00007FF8D0FD31EC  call        qword ptr [Pointer to: System.ArgumentNullException..ctor(System.String) (07FF92D26FD20h)]  
00007FF8D0FD31F2  mov         rcx,rsi  
00007FF8D0FD31F5  call        00007FF92EA285B0  
00007FF8D0FD31FA  int         3  
00007FF8D0FD31FB  add         byte ptr [rcx],bl  
00007FF8D0FD31FD  or          byte ptr [7FF8D53F3A03h],al  
00007FF8D0FD3203  xor         byte ptr [rbx],al  
00007FF8D0FD3205  push        rax  
00007FF8D0FD3206  add         ah,byte ptr [rax+1]  
00007FF8D0FD3209  jo          Microsoft.SharePoint.Client.ClientRuntimeContext.AddQueryIdAndResultObject(Int64, System.Object)+0DBh (07FF8D0FD320Bh)  
00007FF8D0FD320B  add         byte ptr [rax],al  
00007FF8D0FD320E  add         byte ptr [rax],al  
00007FF8D0FD3210  add         byte ptr [rax],al  
00007FF8D0FD3212  add         byte ptr [rax],al  
00007FF8D0FD3214  add         byte ptr [rax],al  
00007FF8D0FD3216  add         byte ptr [rax],al  
00007FF8D0FD3218  nop  
00007FF8D0FD3219  in          al,27h  
00007FF8D0FD321B  sar         eax,1  
00007FF8D0FD321D  jg          Microsoft.SharePoint.Client.ClientRuntimeContext.AddQueryIdAndResultObject(Int64, System.Object)+0EFh (07FF8D0FD321Fh)  
00007FF8D0FD321F  ?? ?????? 
Microsoft.SharePoint.Client.Runtime.dll!Microsoft.SharePoint.Client.ClientRequest.AddQueryIdAndResultObject(long, object):
00007FF8D0FD3220  push        rdi  
00007FF8D0FD3221  push        rsi  
00007FF8D0FD3222  push        rbp  
00007FF8D0FD3223  push        rbx  
00007FF8D0FD3224  sub         rsp,28h  
00007FF8D0FD3228  mov         rsi,rcx  
00007FF8D0FD322B  mov         rbx,rdx  
00007FF8D0FD322E  mov         rdi,r8  
00007FF8D0FD3231  cmp         dword ptr [rsi+60h],0  
00007FF8D0FD3235  jne         Microsoft.SharePoint.Client.ClientRequest.AddQueryIdAndResultObject(Int64, System.Object)+0115h (07FF8D0FD3335h)  
00007FF8D0FD323B  mov         rbp,qword ptr [rsi+58h]  
00007FF8D0FD323F  mov         rcx,1D45F4B19A8h  
00007FF8D0FD3249  mov         rcx,qword ptr [rcx]  
00007FF8D0FD324C  call        System.Globalization.NumberFormatInfo.GetInstance(System.IFormatProvider) (07FF92D5DC510h)  
00007FF8D0FD3251  mov         r8,rax  
00007FF8D0FD3254  mov         rcx,rbx  
00007FF8D0FD3257  xor         edx,edx  
00007FF8D0FD3259  call        00007FF92EA776C0  
00007FF8D0FD325E  mov         rdx,rax  
00007FF8D0FD3261  mov         ecx,dword ptr [rbp]  
00007FF8D0FD3264  mov         rcx,rbp  
00007FF8D0FD3267  mov         r8,rdi  
00007FF8D0FD326A  xor         r9d,r9d  
00007FF8D0FD326D  call        System.Collections.Generic.Dictionary`2[[System.__Canon, mscorlib],[System.__Canon, mscorlib]].Insert(System.__Canon, System.__Canon, Boolean) (07FF92D5D9FC0h)  
00007FF8D0FD3272  mov         rdx,rdi  
00007FF8D0FD3275  mov         rcx,7FF8CF73E768h  
00007FF8D0FD327F  call        00007FF92EA8D9B0  
00007FF8D0FD3284  mov         rdi,rax  
00007FF8D0FD3287  test        rdi,rdi  
00007FF8D0FD328A  je          Microsoft.SharePoint.Client.ClientRequest.AddQueryIdAndResultObject(Int64, System.Object)+010Bh (07FF8D0FD332Bh)  
00007FF8D0FD3290  mov         rcx,7FF8D124DB38h  
00007FF8D0FD329A  mov         rdx,7FF8CF73E608h  
00007FF8D0FD32A4  mov         r8,7FF8CF73E768h  
00007FF8D0FD32AE  mov         r9d,1  
00007FF8D0FD32B4  call        00007FF92ED85940  
00007FF8D0FD32B9  mov         rcx,qword ptr [rdi+10h]  
00007FF8D0FD32BD  cmp         qword ptr [rcx+38h],0  
00007FF8D0FD32C2  je          Microsoft.SharePoint.Client.ClientRequest.AddQueryIdAndResultObject(Int64, System.Object)+010Bh (07FF8D0FD332Bh)  
00007FF8D0FD32C4  mov         rsi,qword ptr [rsi+58h]  
00007FF8D0FD32C8  mov         rcx,1D45F4B19A8h  
00007FF8D0FD32D2  mov         rbp,qword ptr [rcx]  
00007FF8D0FD32D5  mov         rcx,7FF8D124DB38h  
00007FF8D0FD32DF  mov         rdx,7FF8CF73E608h  
00007FF8D0FD32E9  mov         r8,7FF8CF73E768h  
00007FF8D0FD32F3  mov         r9d,1  
00007FF8D0FD32F9  call        00007FF92ED85940  
00007FF8D0FD32FE  mov         rcx,rbp  
00007FF8D0FD3301  call        System.Globalization.NumberFormatInfo.GetInstance(System.IFormatProvider) (07FF92D5DC510h)  
00007FF8D0FD3306  mov         r8,rax  
00007FF8D0FD3309  mov         rcx,rbx  
00007FF8D0FD330C  xor         edx,edx  
00007FF8D0FD330E  call        00007FF92EA776C0  
00007FF8D0FD3313  mov         rdx,rax  
00007FF8D0FD3316  mov         rcx,qword ptr [rdi+10h]  
00007FF8D0FD331A  mov         r8,qword ptr [rcx+38h]  
00007FF8D0FD331E  mov         ecx,dword ptr [rsi]  
00007FF8D0FD3320  mov         rcx,rsi  
00007FF8D0FD3323  xor         r9d,r9d  
00007FF8D0FD3326  call        System.Collections.Generic.Dictionary`2[[System.__Canon, mscorlib],[System.__Canon, mscorlib]].Insert(System.__Canon, System.__Canon, Boolean) (07FF92D5D9FC0h)  
00007FF8D0FD332B  nop  
00007FF8D0FD332C  add         rsp,28h  
00007FF8D0FD3330  pop         rbx  
00007FF8D0FD3331  pop         rbp  
00007FF8D0FD3332  pop         rsi  
00007FF8D0FD3333  pop         rdi  
00007FF8D0FD3334  ret  
00007FF8D0FD3335  mov         rcx,7FF8D124DB38h  
00007FF8D0FD333F  mov         rdx,7FF8D124E788h  
00007FF8D0FD3349  mov         r8,7FF8D124E7B0h  
00007FF8D0FD3353  mov         r9d,1  
00007FF8D0FD3359  call        00007FF92ED85940  
00007FF8D0FD335E  mov         rcx,7FF92D123878h  
00007FF8D0FD3368  call        CORINFO_HELP_NEWSFAST (07FF92EA8FFB0h)  
00007FF8D0FD336D  mov         rsi,rax  
00007FF8D0FD3370  mov         rcx,1D45FC224F0h  
00007FF8D0FD337A  mov         rdi,qword ptr [rcx]  
00007FF8D0FD337D  mov         ecx,4FDh  
00007FF8D0FD3382  mov         rdx,7FF8CF73CBB8h  
00007FF8D0FD338C  call        00007FF92EA87F00  
00007FF8D0FD3391  mov         rdx,rax  
00007FF8D0FD3394  mov         rcx,rdi  
00007FF8D0FD3397  mov         rax,qword ptr [rdi]  
00007FF8D0FD339A  mov         rax,qword ptr [rax+48h]  
00007FF8D0FD339E  call        qword ptr [rax+20h]  
00007FF8D0FD33A1  mov         rdx,rax  
00007FF8D0FD33A4  mov         rcx,rsi  
00007FF8D0FD33A7  call        System.InvalidOperationException..ctor(System.String) (07FF92DE69940h)  
00007FF8D0FD33AC  mov         rcx,rsi  
00007FF8D0FD33AF  call        00007FF92EA285B0  
00007FF8D0FD33B4  int         3  
00007FF8D0FD33B5  add         byte ptr [rax],al  
00007FF8D0FD33B7  add         byte ptr [rcx],bl  
00007FF8D0FD33B9  or          byte ptr [7FF8D53F3BBFh],al  
00007FF8D0FD33BF  xor         byte ptr [rbx],al  
00007FF8D0FD33C1  push        rax  
00007FF8D0FD33C2  add         ah,byte ptr [rax+1]  
00007FF8D0FD33C5  jo          Microsoft.SharePoint.Client.ClientRequest.AddQueryIdAndResultObject(Int64, System.Object)+01A7h (07FF8D0FD33C7h)  
00007FF8D0FD33C7  add         byte ptr [rax],al  
00007FF8D0FD33CA  add         byte ptr [rax],al  
00007FF8D0FD33CC  add         byte ptr [rax],al  
00007FF8D0FD33CE  add         byte ptr [rax],al  
00007FF8D0FD33D0  add         byte ptr [rax],al  
00007FF8D0FD33D2  add         byte ptr [rax],al  
00007FF8D0FD33D4  add         byte ptr [rax],al  
00007FF8D0FD33D6  add         byte ptr [rax],al  
00007FF8D0FD33D8  and         ch,ah  
00007FF8D0FD33DA  ?? ?????? 
00007FF8D0FD33DB  sar         eax,1  
00007FF8D0FD33DD  jg          Microsoft.SharePoint.Client.ClientRequest.AddQueryIdAndResultObject(Int64, System.Object)+01BFh (07FF8D0FD33DFh)  
00007FF8D0FD33DF  ?? ?????? 
Microsoft.SharePoint.Client.Runtime.dll!Microsoft.SharePoint.Client.ClientObject.ClientObject(Microsoft.SharePoint.Client.ClientRuntimeContext, Microsoft.SharePoint.Client.ObjectPath):
00007FF8D0FD33E0  push        rdi  
00007FF8D0FD33E1  push        rsi  
00007FF8D0FD33E2  push        rbx  
00007FF8D0FD33E3  sub         rsp,20h  
00007FF8D0FD33E7  mov         rsi,rcx  
00007FF8D0FD33EA  mov         rdi,r8  
00007FF8D0FD33ED  test        rdx,rdx  
00007FF8D0FD33F0  je          Microsoft.SharePoint.Client.ClientObject..ctor(Microsoft.SharePoint.Client.ClientRuntimeContext, Microsoft.SharePoint.Client.ObjectPath)+077h (07FF8D0FD3457h)  
00007FF8D0FD33F2  lea         rcx,[rsi+8]  
00007FF8D0FD33F6  call        CORINFO_HELP_ASSIGN_REF (07FF92EA8DC60h)  
00007FF8D0FD33FB  mov         rcx,7FF8CF73E598h  
00007FF8D0FD3405  mov         rdx,7FF8D124B248h  
00007FF8D0FD340F  mov         r8,7FF8D124B268h  
00007FF8D0FD3419  mov         r9d,1  
00007FF8D0FD341F  call        00007FF92ED85940  
00007FF8D0FD3424  mov         rcx,7FF8D124B268h  
00007FF8D0FD342E  call        CORINFO_HELP_NEWSFAST (07FF92EA8FFB0h)  
00007FF8D0FD3433  mov         rbx,rax  
00007FF8D0FD3436  mov         rdx,rbx  
00007FF8D0FD3439  lea         rcx,[rsi+10h]  
00007FF8D0FD343D  call        CORINFO_HELP_ASSIGN_REF (07FF92EA8DC60h)  
00007FF8D0FD3442  lea         rcx,[rbx+40h]  
00007FF8D0FD3446  mov         rdx,rdi  
00007FF8D0FD3449  call        CORINFO_HELP_ASSIGN_REF (07FF92EA8DC60h)  
00007FF8D0FD344E  nop  
00007FF8D0FD344F  add         rsp,20h  
00007FF8D0FD3453  pop         rbx  
00007FF8D0FD3454  pop         rsi  
00007FF8D0FD3455  pop         rdi  
00007FF8D0FD3456  ret  
00007FF8D0FD3457  mov         rcx,7FF92D123F90h  
00007FF8D0FD3461  call        CORINFO_HELP_NEWSFAST (07FF92EA8FFB0h)  
00007FF8D0FD3466  mov         rsi,rax  
00007FF8D0FD3469  mov         ecx,1  
00007FF8D0FD346E  mov         rdx,7FF8CF73CBB8h  
00007FF8D0FD3478  call        00007FF92EA87F00  
00007FF8D0FD347D  mov         rdx,rax  
00007FF8D0FD3480  mov         rcx,rsi  
00007FF8D0FD3483  call        qword ptr [Pointer to: System.ArgumentNullException..ctor(System.String) (07FF92D26FD20h)]  
00007FF8D0FD3489  mov         rcx,rsi  
00007FF8D0FD348C  call        00007FF92EA285B0  
00007FF8D0FD3491  int         3  
00007FF8D0FD3492  add         byte ptr [rax],al  
00007FF8D0FD3494  sbb         dword ptr [rdi],eax  
00007FF8D0FD3496  add         al,0  
00007FF8D0FD3498  ?? ?????? 
00007FF8D0FD3499  xor         al,byte ptr [rbx]  
00007FF8D0FD349B  xor         byte ptr [rdx],al  
00007FF8D0FD349D  ?? ?????? 
00007FF8D0FD349E  add         dword ptr [rax+40h],esi  
00007FF8D0FD34A1  add         byte ptr [rax],al  
00007FF8D0FD34A3  add         byte ptr [rax],al  
00007FF8D0FD34A5  add         byte ptr [rax],al  
00007FF8D0FD34A7  add         byte ptr [rax-72ED81Bh],dl  
00007FF8D0FD34AD  jg          Microsoft.SharePoint.Client.ClientObject..ctor(Microsoft.SharePoint.Client.ClientRuntimeContext, Microsoft.SharePoint.Client.ObjectPath)+0CFh (07FF8D0FD34AFh)  
00007FF8D0FD34AF  ?? ?????? 
Microsoft.SharePoint.Client.dll!Microsoft.SharePoint.Client.RequestContext.Site.get():
00007FF8D0FD34B0  push        r14  
00007FF8D0FD34B2  push        rdi  
00007FF8D0FD34B3  push        rsi  
00007FF8D0FD34B4  push        rbp  
00007FF8D0FD34B5  push        rbx  
00007FF8D0FD34B6  sub         rsp,20h  
00007FF8D0FD34BA  mov         rsi,rcx  
00007FF8D0FD34BD  mov         rcx,7FF8D124A288h  
00007FF8D0FD34C7  mov         rdx,7FF8CF73E608h  
00007FF8D0FD34D1  mov         r8,7FF8CF73E768h  
00007FF8D0FD34DB  mov         r9d,1  
00007FF8D0FD34E1  call        00007FF92ED85940  
00007FF8D0FD34E6  mov         rcx,7FF8D124A288h  
00007FF8D0FD34F0  mov         rdx,7FF8D124B168h  
00007FF8D0FD34FA  mov         r8,7FF8D124B268h  
00007FF8D0FD3504  mov         r9d,1  
00007FF8D0FD350A  call        00007FF92ED85940  
00007FF8D0FD350F  mov         rdi,qword ptr [rsi+10h]  
00007FF8D0FD3513  cmp         qword ptr [rdi+18h],0  
00007FF8D0FD3518  jne         Microsoft.SharePoint.Client.RequestContext.get_Site()+095h (07FF8D0FD3545h)  
00007FF8D0FD351A  mov         rcx,7FF92D1075E8h  
00007FF8D0FD3524  call        CORINFO_HELP_NEWSFAST (07FF92EA8FFB0h)  
00007FF8D0FD3529  mov         rbx,rax  
00007FF8D0FD352C  mov         rcx,rbx  
00007FF8D0FD352F  xor         edx,edx  
00007FF8D0FD3531  xor         r8d,r8d  
00007FF8D0FD3534  call        System.Collections.Generic.Dictionary`2[[System.__Canon, mscorlib],[System.__Canon, mscorlib]]..ctor(Int32, System.Collections.Generic.IEqualityComparer`1<System.__Canon>) (07FF92D5D9850h)  
00007FF8D0FD3539  lea         rcx,[rdi+18h]  
00007FF8D0FD353D  mov         rdx,rbx  
00007FF8D0FD3540  call        CORINFO_HELP_ASSIGN_REF (07FF92EA8DC60h)  
00007FF8D0FD3545  mov         rdi,qword ptr [rdi+18h]  
00007FF8D0FD3549  mov         edx,dword ptr [rdi]  
00007FF8D0FD354B  mov         rdx,1D45F927D98h  
00007FF8D0FD3555  mov         rdx,qword ptr [rdx]  
00007FF8D0FD3558  mov         rcx,rdi  
00007FF8D0FD355B  call        System.Collections.Generic.Dictionary`2[[System.__Canon, mscorlib],[System.__Canon, mscorlib]].FindEntry(System.__Canon) (07FF92D617140h)  
00007FF8D0FD3560  test        eax,eax  
00007FF8D0FD3562  jl          Microsoft.SharePoint.Client.RequestContext.get_Site()+0F1h (07FF8D0FD35A1h)  
00007FF8D0FD3564  mov         rcx,qword ptr [rdi+10h]  
00007FF8D0FD3568  cmp         eax,dword ptr [rcx+8]  
00007FF8D0FD356B  jae         Microsoft.SharePoint.Client.RequestContext.get_Site()+0303h (07FF8D0FD37B3h)  
00007FF8D0FD3571  movsxd      rdx,eax  
00007FF8D0FD3574  lea         rdx,[rdx+rdx*2]  
00007FF8D0FD3578  mov         rdx,qword ptr [rcx+rdx*8+18h]  
00007FF8D0FD357D  mov         rdi,rdx  
00007FF8D0FD3580  test        rdi,rdi  
00007FF8D0FD3583  je          Microsoft.SharePoint.Client.RequestContext.get_Site()+0ECh (07FF8D0FD359Ch)  
00007FF8D0FD3585  mov         rcx,7FF8D1249B58h  
00007FF8D0FD358F  cmp         qword ptr [rdi],rcx  
00007FF8D0FD3592  je          Microsoft.SharePoint.Client.RequestContext.get_Site()+0ECh (07FF8D0FD359Ch)  
00007FF8D0FD3594  call        00007FF92EA8DA50  
00007FF8D0FD3599  mov         rdi,rax  
00007FF8D0FD359C  jmp         Microsoft.SharePoint.Client.RequestContext.get_Site()+02F5h (07FF8D0FD37A5h)  
00007FF8D0FD35A1  mov         rcx,7FF8D124A288h  
00007FF8D0FD35AB  mov         rdx,7FF8CF73E5A8h  
00007FF8D0FD35B5  mov         r8,7FF8CF73E768h  
00007FF8D0FD35BF  mov         r9d,1  
00007FF8D0FD35C5  call        00007FF92ED85940  
00007FF8D0FD35CA  mov         rcx,7FF8D124A288h  
00007FF8D0FD35D4  mov         rdx,7FF8CF73E5A8h  
00007FF8D0FD35DE  mov         r8,7FF8CF73E768h  
00007FF8D0FD35E8  mov         r9d,1  
00007FF8D0FD35EE  call        00007FF92ED85940  
00007FF8D0FD35F3  mov         rdi,qword ptr [rsi+8]  
00007FF8D0FD35F7  mov         rbx,rdi  
00007FF8D0FD35FA  mov         rcx,7FF8D124A288h  
00007FF8D0FD3604  mov         rdx,7FF8CF73E5D8h  
00007FF8D0FD360E  mov         r8,7FF8CF73E768h  
00007FF8D0FD3618  mov         r9d,1  
00007FF8D0FD361E  call        00007FF92ED85940  
00007FF8D0FD3623  mov         rbp,qword ptr [rsi+10h]  
00007FF8D0FD3627  mov         rcx,7FF8D124A288h  
00007FF8D0FD3631  mov         rdx,7FF8D124B4E0h  
00007FF8D0FD363B  mov         r8,7FF8D124B528h  
00007FF8D0FD3645  mov         r9d,1  
00007FF8D0FD364B  call        00007FF92ED85940  
00007FF8D0FD3650  mov         rcx,7FF8D124B528h  
00007FF8D0FD365A  call        CORINFO_HELP_NEWSFAST (07FF92EA8FFB0h)  
00007FF8D0FD365F  mov         r14,rax  
00007FF8D0FD3662  mov         r8,qword ptr [rbp+40h]  
00007FF8D0FD3666  mov         rcx,r14  
00007FF8D0FD3669  mov         rdx,rdi  
00007FF8D0FD366C  mov         r9d,1  
00007FF8D0FD3672  call        Microsoft.SharePoint.Client.ObjectPath..ctor(Microsoft.SharePoint.Client.ClientRuntimeContext, Microsoft.SharePoint.Client.ObjectPath, Boolean) (07FF8D0FD2240h)  
00007FF8D0FD3677  mov         rdx,1D45F927D98h  
00007FF8D0FD3681  mov         rdx,qword ptr [rdx]  
00007FF8D0FD3684  lea         rcx,[r14+30h]  
00007FF8D0FD3688  call        CORINFO_HELP_ASSIGN_REF (07FF92EA8DC60h)  
00007FF8D0FD368D  mov         rcx,7FF8D124A288h  
00007FF8D0FD3697  mov         rdx,7FF8D12491A0h

**Comment by Dr-Oli** [comment]
Problem resolved. ClientContext is not thread-safe and we forgot on 2 places to properly create context.
If you are creating conext from existing one use ClientContext.Clone (URI).
